test/scripts/install-kind.sh

```diff
@@ -10,11 +10,8 @@ sudo mv ./kubectl /usr/local/bin/kubectl
 # Install e2e test binary and ginkgo
 # The e2e binaries are published upstream so the ecosystem can consume then directly
 # xref https://github.com/kubernetes/enhancements/blob/master/keps/sig-testing/20190118-breaking-apart-the-kubernetes-test-tarball.md
-# Current e2e binary version for CI is kubernetes 1.20 with the fix for IPv6 network policies #96856
 # We can always get the latest version published from the URL dl.k8s.io/ci/latest.txt
-# curl -L dl.k8s.io/ci/latest.txt
-# v1.21.0-alpha.0.341+46d481b4556e33
-curl -LO https://dl.k8s.io/ci/v1.21.0-alpha.0.341+46d481b4556e33/kubernetes-test-linux-amd64.tar.gz
+curl -LO https://dl.k8s.io/ci/$(curl -L dl.k8s.io/ci/latest.txt)/kubernetes-test-linux-amd64.tar.gz
```

Comment:
we can't do this
we run with a hardcoded kubernetes versions, this can break easily or introduce new tests without us knowing, or we feature we don't have

Reply:
we either need a url that consistently points to a stable 1.20 version, or we need to build from scratch like we used to. We cant just pin to a URL that gets reaped eventually pins to a specific version that wont receive updates (like fixes to tests).

Reply:
use this one https://dl.k8s.io/release/v1.20.4/kubernetes-test-linux-amd64.tar.gz 
Once we bump the kind image version used for testing we bump the e2e to keep them in sync